### PR TITLE
Change actual objective progress of objective detail

### PR DESCRIPTION
**Pull request by nvs266**
### Summary:
Change actual objective progress of objective detail

### Link Issue:
https://github.com/framgia/fgoal-android/issues/132
https://github.com/framgia/fgoal-android/issues/133
https://github.com/framgia/fgoal-android/issues/134
https://github.com/framgia/fgoal-android/issues/135
https://github.com/framgia/fgoal-android/issues/136
https://github.com/framgia/fgoal-android/issues/137
https://github.com/framgia/fgoal-android/issues/138

### CODE REVIEW

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    @Override
+    protected void onStopped() {
+        mPresenter.onStop();
+    }
+
+    @Override
+    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+        String progressText = progress + "%";
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java]: Dùng String format

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    @Override
+    public void onStopTrackingTouch(SeekBar seekBar) {
+
+    }
+
+    @Override
+    public void notifyActualProgressChangedSuccess(double newProgress) {
+        Toast.makeText(getActivity(), R.string.msg_progress_updated_successful,
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java]: Dùng Navigator Class

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    /**
+     * Message for log
+     *
+     * @param message
+     */
+    @Override
+    public void notifyActualProgressChangedFailure(String message) {
+        Toast.makeText(getActivity(), R.string.msg_progress_updated_failure,
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java]: Dùng Navigator Class

File: app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java [hunk 4]
```diff
@@ -136,7 +131,12 @@ public ObjectiveRemoteDataSource(Api api) {
     }
 
     @Override
-    public Observable<Boolean> linkObjective(String  idObjective, String targetIdObjective) {
+    public Observable<Boolean> linkObjective(String idObjective, String targetIdObjective) {
+        return Observable.just(true).delay(1, TimeUnit.SECONDS);
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java]: hardcode ^^

Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java]: (hehe)

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java (hunk 3 above)
Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java]: (resolved)

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java (hunk 2 above)
Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java]: (resolved)

File: app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java (hunk 4 above)
Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java]: (resolved)

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java (hunk 1 above)
Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectiveDialogFragment.java]: (resolved)

File: app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java (hunk 4 above)
Comment (DoanVanToan) [reply app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java]: nên đổi thành objectiveId and targetObjectiveId

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectivePresenter.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * Created by fs-sournary.
+ * Date: 6/28/18.
+ * Description:
+ */
+public class NewActualProgressObjectivePresenter
+        implements NewActualProgressObjectiveContract.Presenter {
+
+    private static final String TAG = "AMEN";
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectivePresenter.java]: tag?

File: app/src/main/java/com/framgia/fgoal/util/Constant.java [hunk 6]
```diff
@@ -66,6 +66,10 @@
 
     public static final String EXTRA_TITLE = "TITLES";
 
+    /**
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fgoal/util/Constant.java]: remove comment

Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/util/Constant.java]: (resolved)

File: app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectivePresenter.java (hunk 5 above)
Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/screen/newactualprogressobjective/NewActualProgressObjectivePresenter.java]: (resolved)

File: app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java (hunk 4 above)
Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/data/source/remote/ObjectiveRemoteDataSource.java]: (resolved)

File: app/src/main/java/com/framgia/fgoal/data/source/ObjectiveDataSource.java [hunk 7]
```diff
@@ -34,7 +37,9 @@
 
         Observable<Boolean> updateContentKeyResult(String newContent, String id);
 
-        Observable<Boolean> linkObjective(String idObjective, String targetIdObjective);
+        Observable<Boolean> linkObjective(String objectiveId, String targetObjectiveId);
+
+        Observable<Boolean> updateActualProgressObjectiveOrKeyResult(String id, double newProgress);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fgoal/data/source/ObjectiveDataSource.java]: rename to updateObjectiveProgress

Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/data/source/ObjectiveDataSource.java]: @DoanVanToan  Hiện tại đang có 2 loại progress nên em để vậy để phân biệt ạ

Comment (DoanVanToan) [reply app/src/main/java/com/framgia/fgoal/data/source/ObjectiveDataSource.java]: estimate thì đâu có change được đâu em

Comment (nvs266) [reply app/src/main/java/com/framgia/fgoal/data/source/ObjectiveDataSource.java]: @DoanVanToan  À vâng ạ :">

**Pull request merged**